src/components/Navbar/Navbar.jsx

```diff
@@ -1,25 +1,24 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import { Link, useLocation } from "react-router-dom";
+import PropTypes from "prop-types";
 import clsx from "clsx";
 import styles from "./navbar.module.scss";
-import { routes } from "../../routes/index";
+import { useAppRoutes } from "../../hooks/useAppRoutes";
 import { Menu } from "./Menu";
+import { Logo } from "../Logo/Logo";
 
-let hamburgerClasses = styles.hamburger;
+const Hamburger = ({ isOpen, handleHamburgerClick }) => {
+  const hamburgerClasses = clsx(styles.hamburger, {
+    [styles.isActive]: isOpen,
+  });
 
-const Hamburger = () => {
-  const [change, setChange] = useState(false);
   const hamburgerClick = () => {
-    setChange(!change);
-    if (change === false) {
-      hamburgerClasses = clsx([styles.hamburger, styles.isActive]);
-    } else {
-      setChange(!change);
-      hamburgerClasses = styles.hamburger;
-    }
+    handleHamburgerClick(!isOpen);
   };
+
   return (
     <button
-      className={hamburgerClasses}
+      className={isOpen ? hamburgerClasses : styles.hamburger}
```

Comment:
Why not pass here only `hamburgerClasses`? You have `clsx` that manipulate styles depending on props value.